app/views/smart_proxies/plugins/_puppet.html.erb

```diff
@@ -0,0 +1,29 @@
+<% dashboard = Dashboard::Data.new("puppetmaster = #{@smart_proxy.name}") %>
+<% @hosts = dashboard.hosts %>
+<% @report = dashboard.report %>
+<h3><%= _("Puppet") %></h3>
+<%= show_feature_version('puppet') %>
+<div class="col-md-2"><%= _('Hosts managed:') %></div>
+<div class="col-md-10">
+  <%= link_to @hosts.length, hosts_path(:search => "puppetmaster = #{@smart_proxy.name}") %>
```

Comment:
if the puppet master name has spaces in it, you would need to escape quotas, this is true for all other searches that we already included before /cc @shlomizadok